### PR TITLE
S2741: dynamically add target upload columns to search results

### PR DESCRIPTION
**Pull request by hjeeves**

### CODE REVIEW

File: caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js [hunk 1]
```diff
@@ -1622,25 +1771,94 @@
...
-      if (
-        inputFile.length > 0 &&
-        !inputFile.prop('disabled') &&
-        inputFile.val() !== ''
-      ) {
+      //if (
+      //  inputFile.length > 0 &&
+      //  !inputFile.prop('disabled') &&
+      //  inputFile.val() !== ''
+      //) {
```
Comment (at88mph) [inline caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js:1851]: The comments can be deleted.

File: caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js [hunk 2]
```diff
@@ -1685,9 +1903,15 @@
...
         try {
+          var formFieldColumns = []
+          if (this.hasInputFile()) {
+            formFieldColumns.push('targetList')
+          }
           this.$form
             .find('input.' + this.configuration.getName() + '_selectlist')
-            .val(this.configuration.getSelectListString(false))
+            .val(this.getSelectListString(false, formFieldColumns))
+            //.val(this.configuration.getSelectListString(false, formFieldColumns))
```
Comment (at88mph) [inline caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js:1913]: Remove comments.

File: caom2-search-server/src/test/java/ca/nrc/cadc/caom2/ADQLGeneratorTest.java [hunk 3]
```diff
@@ -380,9 +380,10 @@ public void getFromWithUploadCAOM() throws Exception
 
         final String fromClause = getTestSubject().getFrom(Plane.class, 2);
 
+        System.out.println("BLAAAAA!!!!!" + fromClause);
```
Comment (at88mph) [inline caom2-search-server/src/test/java/ca/nrc/cadc/caom2/ADQLGeneratorTest.java:383]: This should be removed.

File: caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.app.js [hunk 4]
```diff
@@ -822,6 +823,12 @@
               // Save viewer state from previous search
               preserveColumnState = true
               prevColumns = resultsVOTV.getColumns()
+
+              // Note: known bug here where additional form field columns
+              //  are retained  if the collections set is the same,
+              //  In future, this is the best point in the code to
+              //  remove any formField columns that do not currrently apply
+              // HJ, June 2020
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.app.js:836]: opened issue for this, added reference in code: https://github.com/opencadc/caom2ui/issues/182

File: caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js (hunk 1 above)
Comment (hjeeves) [reply caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js]: done

File: caom2-search-server/src/test/java/ca/nrc/cadc/caom2/ADQLGeneratorTest.java (hunk 3 above)
Comment (hjeeves) [reply caom2-search-server/src/test/java/ca/nrc/cadc/caom2/ADQLGeneratorTest.java]: whoops! done.

File: caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js (hunk 2 above)
Comment (hjeeves) [reply caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js]: done

**Pull request merged**